src/modules/image/providers/placeholder.ts

```diff
@@ -0,0 +1,77 @@
+import type { Faker } from '../../..';
+
+/**
+ * Module to generate links to images on `https://via.placeholder.com/`.
+ */
+export class Placeholder {
+  constructor(private readonly faker: Faker) {}
+
+  /**
+   * Generates a new placeholder image url.
+   *
+   * @param width The width of the image (in pixels). Defaults to `640`.
+   * @param height The height of the image (in pixels). Defaults to `width`.
+   * @param text The text of the image.
+   * @param format The file format of the image. Supports `png` `jpeg` `png` `gif` `webp`.
+   * @param backgroundColor The background color of the placeholder. Supports HEX CODE format.
+   * @param textColor The text color of the placeholder. Requires `backgroundColor` Supports HEX CODE format.
+   */
+  imageUrl(
+    width?: number,
+    height?: number,
+    text?: string,
+    format?: 'png' | 'jpeg' | 'jpg' | 'gif' | 'webp',
+    backgroundColor?: string,
+    textColor?: string
+  ): string {
+    width = width || 640;
+    height = height || width;
+
+    let url = 'https://via.placeholder.com';
+    url += `/${width}/${height}`;
+
+    if (backgroundColor != null) {
+      url += `/${backgroundColor.replace('#', '').toUpperCase()}`;
+    }
+
+    if (textColor != null && backgroundColor !== null) {
+      url += `/${textColor.replace('#', '').toUpperCase()}`;
+    }
```

Comment:
IMO if you write it like this, then it's easier to understand:

```suggestion
    if (backgroundColor != null) {
      url += `/${backgroundColor.replace('#', '').toUpperCase()}`;

      if (textColor != null) {
        url += `/${textColor.replace('#', '').toUpperCase()}`;
      }
    }
```